cql3/type_json.cc

```diff
@@ -75,6 +77,82 @@ static sstring quote_json_string(const sstring& value) {
     return oss.str();
 }
 
+template <std::floating_point Tf, std::integral Ti>
+static std::partial_ordering cmp(Tf x, Ti y) {
+    return x <=> y;
+}
+
+// With IEEE-754, a 64-bit float point number is represented with
+// - 1 sign bit
+// - 11 bits of exponent
+// - 52 bits of significand
+//
+// At the two ends of the range of the numbers represented by a 64 bit float
+// point number, the exponent is greater or equal to 1, so the float point numbers
+// are not able to represent all integers any more. Specifically, two nearest
+// representable integers greater than 2^53 would be have a spacing greater or
+// equal to 2. And, we cannot assume the existence of float128, we are not able
+// to fit an uint64_t or a 64 bit float into a wider numeric type for comparing
+// them, we need to develop an algorithem to tackle the problem of comparing a
+// float point number and a fixed point number, where the fixed point number is
+// so large that the former is not able to represent it accurately, because the
+// number of bits of float point number's significand is not enough for represent
+// the all the bits of a two's complement number.
+//
+// here is a hypothetical example, assuming we need to compare 4.0 with 3, and a
+// 64-bit float point number is not able to represent 3, the nearest float point
+// number to "3" is "2". the hypothetical distribution of IEEE-754 float point
```

Comment:
Can you please use a real example? I don't understand your hypothetical example, or even its notation (you start by saying that 4.0 is floating point and 3 is an int, and then you say that 2 (not 2.0) is floating point, I don't understand>).